src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,215 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node<T> node = new Node<>(null,value,null);
+            head = node;
+            tail = node;
+            size++;
+        } else {
+            Node<T> node = new Node<>(tail,value,null);
+            tail.next = node;
+            tail = node;
+            size++;
+        }
+
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> node;
+
+        if (index == 0) {
+            node = new Node<>(null,value,head);
+
+            if (head != null) {
+                node.next = head;
+                head.prev = node;
+            } else {
+                head = node;
+                tail = node;
+            }
+            head = node;
+            size++;
+        } else {
+            if (index == size) {
+                node = new Node<>(tail,value,null);
+                node.prev.next = node;
+                tail = node;
+                size++;
+            } else if (index > size || index < 0) {
+                throw new IndexOutOfBoundsException();
+            } else {
+                node = head;
+                for (int i = 0; i < size; i++) {
+                    if (i == index) {
+                        node = new Node<>(node.prev,value,node);
+                        node.prev.next = node;
+                        break;
+                    }
+                    node = node.next;
+                }
+                size++;
+            }
+
+        }
     }
```

Comment:
Be careful with the logic in the `add(T value, int index)` method. It seems you're not handling the case when the index is greater than the size of the list or less than 0. You should throw an `IndexOutOfBoundsException` in such cases. Consider extracting the logic for checking the index into a separate method, it will help you avoid code duplication.